### DISCUSSION TITLE
installation wont start

### DISCUSSION THREAD

**Discussion by aero9** [ZerBea/hcxdumptool#346] [General]
git clone https://github.com/ZerBea/hcxtools.git
cd hcxtools

make

hcxdumptool.c: In function ‘channel_to_frequency’:hcxdumptool.c:682:7: error: ‘NL80211_BAND_6GHZ’ undeclared (first use in this function)

  case NL80211_BAND_6GHZ:
hcxdumptool.c:682:7: note: each undeclared identifier is reported only once for each function it appears in
hcxdumptool.c: In function ‘set_interface’:
hcxdumptool.c:3831:68: error: ‘NL80211_BAND_6GHZ’ undeclared (first use in this function)
    else if(userband[0] == 'c') ufreq = channel_to_frequency(uband, NL80211_BAND
Makefile:66: recipe for target 'hcxdumptool' failed
[Makefile:66:: command not found
](make: *** [hcxdumptool] Error 1)
__________________

no matter what i do same i get same error 2 days in the raw. i have ubuntu xenial and atheros usb adapter

also i tried this command to make sure i'm not missing some dependencies 
sudo apt-get install pkg-config libcurl4-openssl-dev libssl-dev zlib1g-dev make gcc

then i installed hcxdumptool 5.1.7 deb file from another source its works   but i cant get pmkid's when i run 
sudo hcxdumptool -i interface -o Wi-Fi_PMKIDnew.pcapng  --enable_status=1

**Comment by ZerBea** [comment]
The new WiFi bands bands are defined since Linux kernel 5.10:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/uapi/linux/nl80211.h?h=v5.10.197#n4852
```
/**
 * enum nl80211_band - Frequency band
 * @NL80211_BAND_2GHZ: 2.4 GHz ISM band
 * @NL80211_BAND_5GHZ: around 5 GHz band (4.9 - 5.7 GHz)
 * @NL80211_BAND_60GHZ: around 60 GHz band (58.32 - 69.12 GHz)
 * @NL80211_BAND_6GHZ: around 6 GHz band (5.9 - 7.2 GHz)
 * @NL80211_BAND_S1GHZ: around 900MHz, supported by S1G PHYs
 * @NUM_NL80211_BANDS: number of bands, avoid using this in userspace
 *	since newer kernel versions may support more bands
 */
enum nl80211_band {
	NL80211_BAND_2GHZ,
	NL80211_BAND_5GHZ,
	NL80211_BAND_60GHZ,
	NL80211_BAND_6GHZ,
	NL80211_BAND_S1GHZ,

	NUM_NL80211_BANDS,
};
```

From hcxdumptool 6.3.0 kernel 5.10 is mandatory as mentioned in requirements section:
`* operating system: Linux (recommended: kernel >= 6.4, mandatory: kernel >= 5.10)`

on Debian:
`### Debian (e.g. Kali, Ubuntu) release requirements >= bookworm (testing/Debian 12)`

If you meet these conditions, e.g.:
```
$ uname -r
6.5.5-arch1-1
```

make should find  this enum:
```
$ grep -r NL80211_BAND_6GHZ
usr/include/linux/nl80211.h: * @NL80211_BAND_6GHZ: around 6 GHz band (5.9 - 7.2 GHz)
usr/include/linux/nl80211.h:	NL80211_BAND_6GHZ,
```

and hcxdumptool compiles and installs fine:
```
$ make
cc -O3 -Wall -Wextra -Wpedantic -std=gnu99   -o hcxdumptool hcxdumptool.c -DVERSION_TAG=\"6.3.1-59-g16d98aa\" -DVERSION_YEAR=\"2023\" -DSTATUSOUT -DNMEAOUT

$ sudo make install
install -D -m 0755 hcxdumptool /usr/bin/hcxdumptool

$ hcxdumptool -v
hcxdumptool 6.3.1-59-g16d98aa (C) 2023 ZeroBeat
compiled by gcc 13.2.1
compiled with Linux API headers 6.4.0
compiled with glibc 2.38
```



If you're running an older kernel than 5.10, hcxdumptool v6.2.9 is the last working version.
Some more information is here:
https://github.com/ZerBea/hcxdumptool/issues/324

**Comment by ZerBea** [comment]
By latest commit
https://github.com/ZerBea/hcxdumptool/commit/acb42702f78340b3c323661d3a221a6003d057b0
hcxdumptool should compile on older kernels, but all WiFi 6 features are disabled.

**Comment by ZerBea** [comment]
Can you please comment output of
`$ uname -r`

In your first comment, you mentioned that you're running Ubuntu Xenial. This is a very old distribution version, released on 21 April 2016.

**Comment by aero9** [comment]
uname -r
4.15.0-142-generic , i think the operating system don't support beyond that   


yes distro is old but if i upgraded to 18.04 my laptop will not run smoothly  
Intel® Core™ i5-8265U CPU @ 1.60GHz × 8 
3.7 GiB


sudo apt-get install pkg-config libcurl4-openssl-dev libssl-dev zlib1g-dev make gcc

gcc is already the newest version (4:5.3.1-1ubuntu1).
make is already the newest version (4.1-6).
pkg-config is already the newest version (0.29.1-0ubuntu1).
libcurl4-openssl-dev is already the newest version (7.47.0-1ubuntu2.19).
libssl-dev is already the newest version (1.0.2g-1ubuntu4.20).
zlib1g-dev is already the newest version (1:1.2.8.dfsg-2ubuntu4.3).

**Comment by ZerBea** [comment]
Thanks for that information.
Kernel 4.15.0-142 is outdate and not longer maintained:
https://www.kernel.org/

The same applies to the installed tools, e.g.:
gcc 13.2 -> your version 4:5.3.1
https://gcc.gnu.org/releases.html

libssl-dev 3.1.3
https://gcc.gnu.org/releases.html

Neither latest hcxdumptool nor latest hcxtools will work on this old versions.

**Comment by aero9** [comment]
thanks for the great support , i have 2 choices then either running kali linux from usb to try hcxdumptool or upgrading to bionic beaver . but i prefer kali since my old OS is working fine without major issues.

**Comment by ZerBea** [comment]
Bionic Beaver doesn't make sense, because you run into the same problems.
Latest KALI provide updated drivers and tools, but you should know that you have to configure/customize it by hand. Otherwise you will run into a lot of problems. If you take a look at the bug reports (hcxtools/hcxdumptool) you'll notice that most of this reports came from KALI users. From my point of view, KALI is a good distribution to be used by  experienced Linux users and and experienced penetration testers but it is definitely not a distribution for a beginner.
You also may have noticed that there is e.g. no script to detect running services that take access to the WiFi device and to stop them. That is the user's job, because the user configured the system and only the user know what service should be stopped and what not. 

I'm running Arch Linux (main systems) and Raspbian OS Lite testing (Raspberry Pi). Both distributions  installed, configured and customized by hand.  That ensure that I know what is going on in the background of my systems. That include driver bugs, as reported e.g. here:
https://bugzilla.kernel.org/show_bug.cgi?id=217465
That doesn't include third party drivers (e.g. installed by KALI) which are not part of the official Linux kernel because there are a way to much of these in the wildness (git).
In addition the rolling release environment ensures that I'm always on latest and updated distribution (driver and software).

On the bright side all hcxdumptool/hcxtools are always state of the art (NETLINK, WiFI6, ...).
But on the dark side they are not backward compatible, because I don't want to install outdated software (especially software and drivers) on my systems.

**Comment by aero9** [comment]
> Bionic Beaver doesn't make sense, because you run into the same problems. Latest KALI provide updated drivers and tools, but you should know that you have to configure/customize it by hand. Otherwise you will run into a lot of problems. If you take a look at the bug reports (hcxtools/hcxdumptool) you'll notice that most of this reports came from KALI users. From my point of view, KALI is a good distribution to be used by experienced Linux users and and experienced penetration testers but it is definitely not a distribution for a beginner. You also may have noticed that there is e.g. no script to detect running services that take access to the WiFi device and to stop them. That is the user's job, because the user configured the system and only the user know what service should be stopped and what not.
> 
> I'm running Arch Linux (main systems) and Raspbian OS Lite testing (Raspberry Pi). Both distributions installed, configured and customized by hand. That ensure that I know what is going on in the background of my systems. That include driver bugs, as reported e.g. here: https://bugzilla.kernel.org/show_bug.cgi?id=217465 That doesn't include third party drivers (e.g. installed by KALI) which are not part of the official Linux kernel because there are a way to much of these in the wildness (git). In addition the rolling release environment ensures that I'm always on latest and updated distribution (driver and software).
> 
> On the bright side all hcxdumptool/hcxtools are always state of the art (NETLINK, WiFI6, ...). But on the dark side they are not backward compatible, because I don't want to install outdated software (especially software and drivers) on my systems.


okay thanks for the advice , i'm not going to install  Bionic Beaver to just letting hcxdumptool work . as for kali i have used couple of times with my current laptop . but i tested  backtrack before kali to come to life . so i'm used to these complicated os's although i don't have robust knowledge i just google issues sometimes i'm able to solve them . at this point i will need standard tools to work probably with kali like airmon-ng and airodump-ng along with hcxdump ,meanwhile supporting my  driver and injection. kali usually packed with a lot of tools that not useful  for everyone like you have mentioned . its often overwhelming but i'm okay with that . 

can you please recommend a stable kali version that not tricky at getting things started and support wide models of wifi drivers , it doesn't have to be latest version since my older hardware might not run at all with cutting edge distribution with tendency to hunt all resources. 
i have TP-LINK TL WN722N adapter  chipset(Atheros AR9271 - driver ath9k)

again thanks for the awesome support

**Comment by ZerBea** [comment]
```
hcxdumptool/hcxtools work if the conditions are met:
hcxdumptool : Linux kernel (>= 5.10)
hcxtools: libopenssl (>= 3.0) and libcurl (>= 7.56)
```

All tools compile fine on gcc (13).

As long as your distribution version meet this conditions, everything is fine.

Recommended is Arch Linux or Raspbian OS Lite

Black Arch should do it, too:
https://www.blackarch.org/

As well as KALI
https://www.kali.org/releases/

BTW:
I have an ancient netbook and Arch Linux 32 is working fine
https://www.archlinux32.org/

I have a TP-LINK TL-WN722N, too and noticed several problems on older drivers. Under heavy load, it stops working.

TP LINK replaced the TL-WN722N v1 (Atheros) by v2(Realtek)  and later on by v3 (Realtek):
https://wikidevi.wi-cat.ru/TP-LINK_TL-WN722N_v3
Full monitor mode and full packet injection on v2/3 is added to Linux kernel since kernel 6.3.

**Comment by aero9** [comment]
i have both TP-LINK TL-WN722N v1 and a couple of TP-LINK TL-WN722N v2 adapters .but i had way more success with v1 over the newer t p-link versions in my past experiences , in terms of driver compatibility and other factors like reception when  connecting to regular wifi network . could be because Atheros chipset always worked out of box .
